qa/tasks/ceph_manager.py

```diff
@@ -1231,45 +1248,55 @@ def raw_cluster_cmd(self, *args):
         """
         Start ceph on a raw cluster.  Return count
         """
-        testdir = teuthology.get_testdir(self.ctx)
-        ceph_args = [
-            'sudo',
-            'adjust-ulimits',
-            'ceph-coverage',
-            '{tdir}/archive/coverage'.format(tdir=testdir),
-            'timeout',
-            '120',
-            'ceph',
-            '--cluster',
-            self.cluster,
-        ]
-        ceph_args.extend(args)
-        proc = self.controller.run(
-            args=ceph_args,
-            stdout=StringIO(),
+        if self.ceph_daemon:
+            proc = shell(self.ctx, self.cluster, self.controller,
+                         args=['ceph'] + list(args),
+                         stdout=StringIO())
+        else:
+            testdir = teuthology.get_testdir(self.ctx)
+            ceph_args = [
+                'sudo',
+                'adjust-ulimits',
+                'ceph-coverage',
+                '{tdir}/archive/coverage'.format(tdir=testdir),
+                'timeout',
+                '120',
+                'ceph',
+                '--cluster',
+                self.cluster,
+            ]
+            ceph_args.extend(args)
+            proc = self.controller.run(
```

Comment:
Can we use remote.sh instead of remote.run here?

Reply:
this is the old code path

remote.run is *everywhere* in teuthology, and we are passing various kwargs with things like wait, stdin, stdout, etc.  I'd prefer to leave any run -> sh transition as future work since that is not a problem unique to this new task

Reply:
i thought .sh took a space-separated string insted of args=[..]?  or is that the only difference and the other args are the same?

Reply:
do you have some example code for io.BytesIO? i don't see any instances of this in the tree

Reply:
> i thought .sh took a space-separated string insted of args=[..]? or is that the only difference and the other args are the same?

For py2/py3:
```
proc_obj = remote.run(args=list_or_str, stdin=BytesIO(b'Hi'), stdout=BytesIO(), **otherargs)
output_as_str = str(proc_obj.stdout.getvalue().decode())
```
should be equivalent to:
```
output_as_str = remote.sh(list_or_str, stdin=BytesIO(b'Hi'), **otherargs)
```

Reply:
sorry, i can't make this change without what i'm guessing is your wip branch.. see http://pulpito.ceph.com/sage-2019-11-21_19:08:28-rados:ssh-wip-revert-asio-distro-basic-smithi/.  I'll keep teh patch around but you'll probably have to rebase on top of this after it merges.

Reply:
Oh, sorry me, the cluster object does not have the 'cluster.sh()', I was saying about 'remote.sh'.

Reply:
well, okay, don't mind, will patch it later.